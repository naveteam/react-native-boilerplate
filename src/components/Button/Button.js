import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native'
import styled from 'styled-components'

import { parseModifiers } from 'src/utils'

const ButtonComponent = ({ text, ...props }) => (
  <Button {...props}>
    <Text>{text}</Text>
  </Button>
)

const Button = styled.TouchableOpacity`
  color: black;
  padding: 10px;
  border-radius: 4px;
  background-color: ${props => props.color || 'white'};
  ${props => parseModifiers(props)}
`

ButtonComponent.propTypes = {
  text: PropTypes.string,
  color: PropTypes.string
}

export default ButtonComponent
