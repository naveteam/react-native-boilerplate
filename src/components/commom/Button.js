import React from 'react'
import styled from 'styled-components'

const Button = styled.TouchableOpacity`
  background-color: ${props => props.bg || 'white'};
  color: black;
  padding: 5px;
`

const ButtonText = styled.Text`
  font-size: 20;
  font-weight: bold;
  color: #000;
`

export default ({ bg, name }) => (
  <Button>
    <ButtonText bg={bg}>{name}, Bem vindo</ButtonText>
  </Button>
)
