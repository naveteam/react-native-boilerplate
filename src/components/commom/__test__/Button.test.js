import React from 'react'
import { Text } from 'react-native'
import { create } from 'react-test-renderer'
import 'jest-styled-components'

import Button from '../Button'

describe('<Button />', () => {
  it('render Button with custom name', () => {
    const tree = create(<Button name='João' />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render Button with custom Background color', () => {
    const tree = create(<Button text='João' bg='red' />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
