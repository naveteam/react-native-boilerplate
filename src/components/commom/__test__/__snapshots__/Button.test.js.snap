// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render Button with custom Background color 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "color": "black",
      "opacity": 1,
      "paddingBottom": 5,
      "paddingLeft": 5,
      "paddingRight": 5,
      "paddingTop": 5,
    }
  }
>
  <Text
    bg="red"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    , Bem vindo
  </Text>
</View>
`;

exports[`<Button /> render Button with custom name 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "color": "black",
      "opacity": 1,
      "paddingBottom": 5,
      "paddingLeft": 5,
      "paddingRight": 5,
      "paddingTop": 5,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Jo√£o
    , Bem vindo
  </Text>
</View>
`;
